langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -89,4 +96,22 @@ public static BArray findAllGroups(BRegexpValue regExp, BString str, int startIn
         }
         return groupArray;
     }
+
+    private static void checkIndexWithinRange(BString str, long startIndex) {
+        if (startIndex != (int) startIndex) {
+            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.REGEXP_OPERATION_ERROR,
```

Comment:
Add tests to cover this if not already addded

Reply:
Added in https://github.com/ballerina-platform/ballerina-lang/pull/39238/commits/65cc74722ef47a62b505b5f78ae169435016c70d.